apps/web/lib/settings/organizations/new/getServerSideProps.tsx

```diff
@@ -1,9 +1,24 @@
 import type { GetServerSidePropsResult } from "next";
+import type { GetServerSidePropsContext } from "next";
+import { getSession } from "next-auth/react";
 
 import { getFeatureFlag } from "@calcom/features/flags/server/utils";
 import prisma from "@calcom/prisma";
 
-export const getServerSideProps = async (): Promise<GetServerSidePropsResult<{ isOrg: boolean }>> => {
+export const getServerSideProps = async (
+  ctx: GetServerSidePropsContext
+): Promise<GetServerSidePropsResult<{ isOrg: boolean }>> => {
+  const session = await getSession(ctx);
+
+  if (!session || !session.user) {
+    return {
+      redirect: {
+        destination: `/login`,
+        permanent: false,
+      },
+    };
+  }
```

Comment:
Let's do this directly in RSC in `settings/organizations/new/page.tsx`.

But we should do`redirect('/login')` instead. We can do this before the line `await getData(buildLegacyCtx(await headers(), await cookies(), await params, await searchParams));`

Ideally, we get rid of using `withAppDirSsr` HoC and write other logic directly in RSC as well, but we can move on since that is not in the scope of this PR.